components/prize_pool/commons/starcraft_starcraft2/prize_pool_legacy_starcraft.lua

```diff
@@ -0,0 +1,396 @@
+-- wiki=commons
+-- page=Module:PrizePool/Legacy/Starcraft
+--
+-- Please see https://github.com/Liquipedia/Lua-Modules to contribute
+--
+
+local Array = require('Module:Array')
+local Currency = require('Module:Currency')
+local Json = require('Module:Json')
+local Logic = require('Module:Logic')
+local Lua = require('Module:Lua')
+local String = require('Module:StringUtils')
+local Table = require('Module:Table')
+local Template = require('Module:Template')
+local Variables = require('Module:Variables')
+
+local CustomPrizePool = Lua.import('Module:PrizePool/Custom', {requireDevIfEnabled = true})
+local LegacyPrizePool = Lua.import('Module:PrizePool/Legacy', {requireDevIfEnabled = true})
+local OldStarcraftPrizePool = Lua.import('Module:PrizePool/Starcraft/next', {requireDevIfEnabled = true})
+local Opponent = Lua.import('Module:Opponent', {requireDevIfEnabled = true})
+
+local StarcraftLegacyPrizePool = {}
+
+local AUTOMATION_START_DATE = '2022-01-14'
+local SPECIAL_PLACES = {dq = 'dq', dnf = 'dnf', dnp = 'dnp', w = 'w', d = 'd', l = 'l', q = 'q'}
+
+local CACHED_DATA = {
+	next = {points = 1, qual = 1, freetext = 1},
+	inputToId = {},
+	qualifiers = {},
+}
+
+function StarcraftLegacyPrizePool.run(frame)
+	local args = Template.retrieveReturnValues('PrizePool')
+	local header = Array.sub(args, 1, 1)[1]
+
+	if Logic.readBool(header.award) then
+		OldStarcraftPrizePool.TemplatePrizePoolEnd()
+	end
+
+	local slots = Array.sub(args, 2)
+
+	local newArgs = {}
+
+	newArgs.prizesummary = false
+	newArgs.cutafter = header.cutafter
+	newArgs.lpdb_prefix = header.lpdb_prefix
+
+	if Currency.raw(header.localcurrency) then
+		-- If the localcurrency is a valid currency, handle it like currency
+		newArgs.localcurrency = header.localcurrency
+		CACHED_DATA.inputToId.localprize = 'localprize'
+	else
+		-- Otherwise handle it like it would been a points input using different parameter
+		StarcraftLegacyPrizePool._assignType(newArgs, header.localcurrency, 'localprize')
+	end
+
+	StarcraftLegacyPrizePool._assignType(newArgs, header.points, 'points')
+	StarcraftLegacyPrizePool._assignType(newArgs, header['2points'] or header.points2, 'points2')
+
+	local defaultOpponentType = Opponent.readType(header.opponentType or Opponent.solo)
+	if defaultOpponentType then
+		newArgs.type = {type = defaultOpponentType}
+	end
+	assert(newArgs.type, 'Invalid opponentType set.')
+	newArgs.type.isarchon = header.defaultIsArchon
+
+	CACHED_DATA.defaultOpponentType = defaultOpponentType
+	CACHED_DATA.defaultIsArchon = header.defaultIsArchon
+
+	newArgs.storesmw = header.storeSmw or header['smw mute']
+	newArgs.storelpdb = header.storeLpdb
+	newArgs.storeTournament = header.storeTournament
+	newArgs.series = header.series
+	newArgs.tier = header.tier
+	newArgs['tournament name'] = header['tournament name']
+
+	-- import settings
+	if header.lpdb and header.lpdb ~= 'auto' then
+		newArgs.import = header.lpdb
+	end
+	newArgs.import = StarcraftLegacyPrizePool._enableImport(newArgs)
```

Comment:
This function call needs to be below the `tournament` & `matchGroupId` handling right?